designpatterns/gangoffour/src/main/kotlin/com/felipecosta/designpatterns/creational/Builder.kt

```diff
@@ -0,0 +1,109 @@
+package com.felipecosta.designpatterns.creational
+
+//awful repetition, but this is to avoid clashing with other patterns - I'm open to sugestions.
+
+abstract class MazeBuilder {
+
+    internal val mazeBeingBuilt = BuilderMaze();
+
+    abstract fun makeRoom(number: Int): BuilderRoom;
+    abstract fun makeMazeWall(): BuilderWall;
+    abstract fun makeDoor(outRoom: BuilderRoom, inRoom: BuilderRoom): BuilderDoor;
+
+    fun getResult(): BuilderMaze {
+        return mazeBeingBuilt;
+    }
+}
+
+open class MazeDirector(val builder: MazeBuilder) {
+    fun Construct() {
+        val room1 = builder.makeRoom(1)
+        val room2 = builder.makeRoom(2)
+        val door = builder.makeDoor(room1, room2)
+
+        builder.mazeBeingBuilt.addRoom(room1)
+        builder.mazeBeingBuilt.addRoom(room2)
+
+        room1.setSide(RoomSide.NORTH, builder.makeMazeWall())
+        room1.setSide(RoomSide.EAST, door)
+        room1.setSide(RoomSide.SOUTH, builder.makeMazeWall())
+        room1.setSide(RoomSide.WEST, builder.makeMazeWall())
+        room2.setSide(RoomSide.NORTH, builder.makeMazeWall())
+        room2.setSide(RoomSide.EAST, builder.makeMazeWall())
+        room2.setSide(RoomSide.SOUTH, builder.makeMazeWall())
+        room2.setSide(RoomSide.WEST, door)
+    }
+}
+
+class BuilderDoor(val outRoom: BuilderRoom, val inRoom: BuilderRoom)
```

Comment:
Using a prefix for all of this classes seems strange, they all look like a `Builder` object.
I suggest creating a new package `builder` and moving it into inside.
I will create separated packages for the others patterns as well. :)